Assets/Scripts/BasicsCreator.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class BasicsCreator : MonoBehaviour
```

Comment:
I think this class may be doing a bit much? It seems there are a few different concerns handled here:
- Handling of the UI elements (input fields, buttons, sliders)
- Handling of the 3D space events (mouse click in the empty space, tracking the current screen/mouse position)
- Management of point/segment creation (current mode, tracking of point/segment creation)

I think that the separation of the last part is especially important since other objects 
(Points, Segments) will depend on that, not so much on the handling of other UI/3D space elements.

Separation may also help with a bit more intuitive class names.